### PR TITLE
feat(testcases discovery): Replace unreliable totalnumberoftestsparser with vstest

### PR DESCRIPTION
**Pull request by Mobrockers**
Since we experience a large number of bugs with the TotalNumberOfTestsParser we decided to simply replace it with VsTest. VsTest will now always be used to detect the total number of tests even  if dotnet test is used to run the actual testruns.

### CODE REVIEW

**Comment by dupdob** [comment]
quick remarks: I feel the reason people may have to use dotnet test directly, instead of using VsTest, is a situation where vstest is incompatible for some reason.

I would favor using VsTest as the default runner, instead of dotnet test.

Does that make sense to you?

**Comment by Mobrockers** [comment]
We prefer to move to vstest as the default but I am hesitant to change the default at this time. I'll think about it and reconsider.

File: src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunnerPool.cs [hunk 1]
```diff
@@ -26,10 +26,21 @@ public VsTestRunnerPool(StrykerOptions options, ProjectInfo projectInfo)
             });
         }
 
+        public int DiscoverTests()
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunnerPool.cs]: The name of this method is a bit misleading. It only gets the number of test cases. The current name implies that it does more than that.

File: src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs [hunk 2]
```diff
@@ -130,6 +124,16 @@ public ICollection<TestCase> DiscoverTests(string runSettings = null)
             return _discoveredTests;
         }
 
+        private void SetListOfTests(ICollection<TestCase> tests)
+        {
+            _discoveredTests = tests;
+            if (tests != null)
+            {
+                _withNUnit = tests.Any(testCase => testCase.ExecutorUri.AbsoluteUri.Contains("nunit"));
+                _withXUnit = _discoveredTests.Any(testCase => testCase.Properties.Any(p => p.Id == "XunitTestCase"));
```
[the remark this answers is not in the document]
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs]: Actually, you can have tests from multiple frameworks within the same assembly
this is unusual, but supported!

**Pull request merged**